### ISSUE TITLE
Please allow automatic optimization for multiple optimizers again.

### ISSUE THREAD

**Issue by profPlum** [Lightning-AI/pytorch-lightning#20094]
### Description & Motivation

I'm suggesting allowing the old behavior to work again. While still giving users the option to use the new behavior if they set self.automatic_optimization=False. The old API was well designed and can allow for extremely simple implementations especially in situations where the training step is the same for each optimizer being used (i.e. no optimizer_idx if-statement).

### Pitch

The original primary purpose of Pytorch-Lightning was to simplify & eliminate the boiler plate in the pytorch training loop. But the new behavior is more complicated than even using base pytorch. Since it requires setting flags like self.automatic_optimization=False, then using custom rewrites of well-known base pytorch APIs like loss.backwards(), in addition to reintroducing the boiler-plate that Pytorch-Lightning was made to remove.

### Additional context

Could you at least consider collecting user feedback before you remove useful features like this in the future?

**Comment by awaelchli** [comment]
This will unlikely return as it adds a considerable amount of complexity to the training loops. The other issue was that users had many questions of how the optimizer loop works under the hood. It is true that manual optimization has its issues as well, but it does give the flexibility that users of multiple optimizers often needed, but wasn't accessible in automatic mode. The same argument was applied to the removal of the special TBPTT support.

**Comment by profPlum** [comment]
I'm not suggesting bringing back the entire feature just the basic version as given by the psuedo code below.
If users want different frequencies then they can just repeat the same optimizer multiple times inside the config_optimizers() method (same for lr_schedulers).

And perhaps the code could be organized into some function which handles the optimizer+training_step, that could hide the extra details.

Pseudo code (from pytorch_lightning 1.9.5):
```
for epoch in epochs:
    for batch in data:
        for opt in optimizers:
            loss = train_step(batch, batch_idx, optimizer_idx)
            opt.zero_grad()
            loss.backward()
            opt.step()

    for lr_scheduler in lr_schedulers:
        lr_scheduler.step()
```

**Comment by So-cean** [comment]
hello, i think multiple optimizer in automatic optimization for training GAN-like network is really needed
